device/alpaka/src/utils/queue.cpp

```diff
@@ -0,0 +1,73 @@
+/**
+ * traccc library, part of the ACTS project (R&D line)
+ *
+ * (c) 2025 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Local include(s).
+#include "traccc/alpaka/utils/queue.hpp"
+
+#include "utils.hpp"
+
+// Alpaka include(s).
+#include <alpaka/alpaka.hpp>
+
+namespace traccc::alpaka {
+
+struct queue::impl {
+    /// The device the queue is created for
+    std::size_t m_device{INVALID_DEVICE};
+    /// Bare pointer to the wrapped queue object
+    Queue* m_queue = nullptr;
+    /// Unique pointer to the managed Queue object
+    std::unique_ptr<Queue> m_managedQueue;
+};  // struct queue::impl
+
+queue::queue(std::size_t device) : m_impl{std::make_unique<impl>()} {
+
+    m_impl->m_device = device == INVALID_DEVICE ? 0 : device;
+    m_impl->m_managedQueue = std::make_unique<Queue>(
+        ::alpaka::getDevByIdx(::alpaka::Platform<Acc>{}, m_impl->m_device));
+    m_impl->m_queue = m_impl->m_managedQueue.get();
+}
+
+queue::queue(void* input_queue) : m_impl{std::make_unique<impl>()} {
+
+    assert(input_queue != nullptr);
+    m_impl->m_queue = static_cast<Queue*>(input_queue);
+    m_impl->m_device =
+        ::alpaka::getNativeHandle(::alpaka::getDev(*m_impl->m_queue));
```

Comment:
Argh... I didn't check, I thought the failure was because of an infrastructure issue.

But this is a new bug. I believe `::alpaka::getDev(...)` gets exactly the integer that you're after here. The `::alpaka::getNativeHandle(...)` "surrounding" should be removed.

But of course test it before pushing my suggestion. 😉

Reply:
Looking into this, I think the way I am doing it is correct (`getDev` returns the actual `alpaka::DevUniformCudaHipRt` or `alpaka::DevGenericSycl` etc, rather than the device num)...but only for CUDA + HIP.

For SYCL, it returns the `sycl::device` + `sycl::context` (I guess because the "device" in use is a lot harder to distinguish, with the larger set of devices it could be), so I may just need to special case SYCL to pull out the `sycl::info::device::vendor_id` or whatever the appropriate ID is there from the `sycl::device`.